### ISSUE TITLE
Getting started with data preprocessing in python

### ISSUE THREAD

**Issue by Madrinmarison** [section-engineering-education/engineering-education#2172]
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article
Getting started with data preprocessing in python

### Introduction paragraph (2-3 paragraphs):
Data preprocessing is one of the core steps in machine learning in which data is transformed into an appropriate format that makes it suitable to implement our models.
Data preprocessing involves 6 main steps.
These steps include importing the required libraries, importing the dataset, imputing the missing values, encoding categorical variables, splitting data into the training and testing sets, and finally features scaling.
Imputing missing values: this involves eliminating NA values from the data.
Encoding categorical variables: Involves transforming categorical variables into numeric values.
Splitting data set into training and test set: the training set is the portion of the data that we use to implement our model while the test set is the part we use to evaluate our model no
Feature Scaling: this involves transforming the units of the measurements of the features into standard units.
Feature are simply independent variables.

In this article, I'll explain what these steps are and also how to implement them in python. This will enable us to have appropriate data for future machine learning models.
### Key takeaways: 
1. Reader will know and appreciate the importance of data preprocessing as I'll explain everything as to why it's needed.
2. Reader will able to carry out data preprocessing activities in python.
3. Reader will have a clear understanding of the general activity.

### References: 
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

**Comment by hectorkambow** [comment]
seems like a cool topic @Madrinmarison  - approved

**Comment by hectorkambow** [comment]
@Madrinmarison just wanted to follow up on this topic as I'm clearing up the queue where possible? Can be re-opened when needed.
Of perhaps I missed the PR this is attached to?
